src/main/services/csfloat/csfloatSocket.ts

```diff
@@ -0,0 +1,554 @@
+            );
+
+            const given_ids = trade.given_assets.map(
+                (items_to_give) => items_to_give.asset_id
+            );
+
+            return !![...received_ids, ...given_ids].find((e) => {
+                return assetsToFind[e];
+            });
+        }) as IHistoryPingBody[];
+
+        if (historyForCSFloat.length === 0) return;
+
+        await this._csFloatClient.tradeHistoryStatus(historyForCSFloat);
+    }
+
+    private async pingSentTradeOffers(tradeOffers: IGetTradeOffersResponse
+    ): Promise<void> {
+        const tradeOffersSents = tradeOffers.sent
+        const tradesInPending = await this._csFloatClient.getTrades(
+            EStatusTradeCSFLOAT.PENDING
+        );
+
+        const offersToFind = tradesInPending.reduce((acc, trade) => {
+            acc[trade.steam_offer.id] = true;
+            return acc;
+        }, {} as { [key: string]: boolean });
+
+        const offersForCSFloat = tradeOffersSents.filter((offer) => {
+            return !!offersToFind[offer.id];
+        });
+
+        if (offersForCSFloat.length > 0) {
+            const offersForCSFloatMapped = offersForCSFloat.map((offer) => {
+                return {
+                    offer_id: offer.id,
+                    state: offer.state,
+                    given_asset_ids: offer?.itemsToGive.map(item => item.assetid) || [],
+                    received_asset_ids: offer?.itemsToReceive?.map(item => item.assetid) || [],
+                    time_created: offer.created ? Number(offer.created) : Number(new Date()),
+                    other_steam_id64: offer.partner.getSteamID64(),
+                } as OfferStatus
+            })
+
+            await this._csFloatClient.tradeOffers(offersForCSFloatMapped);
+        }
+
+        for (const offer of tradeOffersSents) {
+            const itemsToGive = offer.itemsToGive.map(
+                (item) => item.assetid
+            ) as string[];
+            const itemsToReceive = offer.itemsToReceive.map(
+                (item) => item.assetid
+            ) as string[];
+
+
+            if (offer.state !== ETradeOfferStateCSFloat.Active) continue;
+
+            const hasTradeWithNoOfferAnnotated = tradesInPending.find((trade) => {
+                if (trade.steam_offer.id) {
+                    return false;
+                }
+
+                return (itemsToGive || []).includes(trade.contract.item.asset_id);
+            });
+
+            if (!hasTradeWithNoOfferAnnotated) continue;
+
+            try {
+                await this._csFloatClient.annotateOffer({
+                    offer_id: offer.id,
+                    given_asset_ids: itemsToGive || [],
+                    received_asset_ids: itemsToReceive || [],
+                    other_steam_id64: offer.partner.getSteamID64(),
+                });
+            } catch (error) {
+                console.error(`failed to annotate offer ${offer.id} post-hoc`, error);
+            }
```

Comment:
Emit the error, so it can be logged
```suggestion
            } catch (error) {
                console.error(`failed to annotate offer ${offer.id} post-hoc`, error);
                this.emit("error", error)
            }
```